src/jemalloc.c

```diff
@@ -930,92 +937,124 @@ malloc_slow_flag_init(void) {
 }
 
 static void
-malloc_conf_init(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS]) {
+malloc_conf_init_helper(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS],
+    bool search_confirm_conf_only, const char *opts_cache[4],
+    char buf[PATH_MAX + 1], const char *empty_opts) {
+	static const char *opts_explain[4] = {
+		"string specified via --with-malloc-conf",
+		"string pointed to by the global variable malloc_conf",
+		"\"name\" of the file referenced by the symbolic link named "
+		    "/etc/malloc.conf",
+		"value of the environment variable MALLOC_CONF"
+	};
 	unsigned i;
-	char buf[PATH_MAX + 1];
 	const char *opts, *k, *v;
 	size_t klen, vlen;
 
 	for (i = 0; i < 4; i++) {
```

Comment:
Nit: now that we have the magic number 4 populated in multiple places, let's define a macro somewhere.